src/platform/nrfconnect/FactoryDataProvider.cpp

```diff
@@ -42,7 +42,15 @@ CHIP_ERROR FactoryDataProvider<FlashFactoryData>::Init()
     uint8_t * factoryData = nullptr;
     size_t factoryDataSize;
 
-    CHIP_ERROR error = mFlashFactoryData.GetFactoryDataPartition(factoryData, factoryDataSize);
+    CHIP_ERROR error = mFlashFactoryData.ProtectFactoryDataPartitionAgainstWrite();
+
+    if (error != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to protect the factory data partition");
+        return error;
```

Comment:
So does this mean that ExternalFlashFactoryData is effectively not supported?

Reply:
Oh, That's my mistake. We support only putting factory data into external storage, not protection against writing. I will change it. Thanks for your watchfulness.